deeppavlov/models/sklearn/sklearn_component.py

```diff
@@ -0,0 +1,357 @@
+                 load_path: Union[str, Path] = None,
+                 infer_method: str = "predict", **kwargs) -> None:
+        """
+        Initialize component with given parameters
+        """
+
+        super().__init__(save_path=save_path, load_path=load_path, **kwargs)
+        self.model_name = model_name
+        self.model_params = kwargs
+        self.model = None
+        self.pipe_params = {}
+        for required in ["in", "out", "fit_on", "main", "name"]:
+            self.pipe_params[required] = self.model_params.pop(required, None)
+
+        self.load()
+        self.infer_method = infer_method
+        self.epochs_done = 0
+        self.batches_seen = 0
+        self.train_examples_seen = 0
+
+    def fit(self, *args, **kwargs) -> None:
+        """
+        Fit model on the given data
+
+        Args:
+            *args: list of x-inputs and, optionally, one y-input (the last one) to fit on.
+                Possible input (x0, ..., xK, y) or (x0, ..., xK) '
+                where K is the number of input data elements (the length of list ``in`` from config). \
+                In case of several inputs (K > 1) input features will be stacked. \
+                For example, one has x0: (n_samples, n_features0), ..., xK: (n_samples, n_featuresK), \
+                then model will be trained on x: (n_samples, n_features0 + ... + n_featuresK).
+            **kwargs: additional parameters
+
+        Returns:
+            None
+        """
+        n_inputs = len(self.pipe_params["in"]) if isinstance(self.pipe_params["in"], list) else 1
+        x_features = self.compose_input_data(args[:n_inputs])
+        if len(args) > n_inputs:
+            y_ = np.squeeze(np.array(args[-1]))
+        else:
+            y_ = None
+
+        try:
+            log.info("Fitting model {}".format(self.model_name))
+            self.model.fit(x_features, y_)
+        except TypeError or ValueError:
+            try:
+                if issparse(x_features):
+                    log.info("Converting input for model {} to dense array".format(self.model_name))
+                    self.model.fit(x_features.todense(), y_)
+                else:
+                    log.info("Converting input for model {} to sparse array".format(self.model_name))
+                    self.model.fit(csr_matrix(x_features), y_)
+            except:
+                raise ConfigError("Can not fit on the given data".format(self.model_name))
+
+        return
+
+    def __call__(self, *args, **kwargs) -> np.ndarray:
+        """
+        Infer on the given data according to given in the config infer method, \
+            e.g. ``"predict", "predict_proba", "transform"``
+
+        Args:
+            *args: list of inputs
+            **kwargs: additional arguments
+
+        Returns:
+            predictions, e.g. list of labels, array of probability distribution, sparse array of vectorized samples
+        """
+        predictions = self.infer_on_batch(args)
```

Comment:
why not just rename `infer_on_batch` to `__call__`?

Reply:
it's a habit after neural networks and wrapping their methods. fixed